.github/workflows/tests.yml

```diff
@@ -21,13 +21,15 @@ jobs:
         - name: install dependencies
           run: |
             sudo apt-get update -y -qq
-            sudo apt-get install -y --allow-downgrades --allow-remove-essential --allow-change-held-packages -qq build-essential gfortran libhdf5-serial-dev
+            sudo apt-get install -y --allow-downgrades --allow-remove-essential --allow-change-held-packages -qq build-essential gfortran libhdf5-serial-dev binutils-gold
```

Comment:
`binutils-gold` gets us use the [gold linker](https://en.wikipedia.org/wiki/Gold_(linker)) developed by google specifically for big C++ projects. It seems to be *much* faster and more memory efficient. Generically I don't know if we want to rely on it, but honestly I was very impressed.

Reply:
I'll defer to @rbberger @dholladay00 and @mauneyc-LANL on their opinions here

Reply:
Yeah i think for now we don't need to change it elsewhere. In this particular CI test, this, GNU ld is too memory inefficient and the test doesn't finish without gold.